fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -1249,11 +1253,18 @@ public LoanScheduleData extractData(@NotNull final ResultSet rs) throws SQLExcep
                 if (fromDate == null) {
                     fromDate = this.lastDueDate;
                 }
-                final BigDecimal outstandingPrincipalBalanceOfLoan = this.outstandingLoanPrincipalBalance.subtract(principalDue);
+
+                BigDecimal outstandingPrincipalBalanceOfLoan = this.outstandingLoanPrincipalBalance.subtract(principalDue);
+                if (isAdditional) {
+                    outstandingPrincipalBalanceOfLoan = this.outstandingLoanPrincipalBalance.add(principalDue);
```

Comment:
Can you help me to understand better why we need to add the principal due amount back?

Reply:
As in the line 1163 above, when we are in the disbursement installment, this is added, so in this line when we have the scenario N+1 (isAdditional in true) we must to add this amount, instead of to have subtracted and probably to get a negative balance